src/frontend/Semantic_check.ml

```diff
@@ -290,9 +290,45 @@ let semantic_check_fn_normal ~is_cond_dist ~loc id es =
     | Some _ ->
         (* Check that Funaps are actually functions *)
         Semantic_error.returning_fn_expected_nonfn_found loc id.name |> error
-    | None ->
-        Semantic_error.returning_fn_expected_undeclaredident_found loc id.name
-        |> error)
+    | None -> (
+        let gen_prefix l = String.concat ~sep:"_" (List.rev l) in
+        match List.rev (String.split id.name ~on:'_') with
+        | (("lpmf" | "lupmf") as suffix) :: tl
+          when Utils.is_distribution_name
```

Comment:
I think this excludes the (deprecated) `cdf_log` forms? Which I think we should include. A test to add:

```stan
model {
    target += von_mises_ccdf_log(1, 0,1);
}
```